### ISSUE TITLE
OuterTypeFilename to allow conventional JUnit 5 classes

### ISSUE THREAD

**Issue by rkrisztian** [checkstyle/checkstyle#13880]
Example code:

```java
class MyTest {
     @Test
     void testSomething() {
         // ...
     }
}
```

With the `OuterTypeFilename` rule, this results in:

```none
Checkstyle: The name of the outer type and the file do not match. 
```

But this is a standard practice with JUnit 5 test classes. The rule `OuterTypeFilename` should detect JUnit 5 test classes (they minimally require at least one test method with the `@Test` annotation, and if it is, the rule should not apply in this case.

For now I think I will just apply a suppression like this:

```xml
<suppress checks="OuterTypeFilename" files="^(?!.*[/\\]main[/\\]).*\.java$"/>
```

But this suppression is too wide. Not every class in the test source sets is a test class. Alternatively I could check for `**/*Test.java`, `**/*IT.java`, etc., whatever naming convention my project uses.

**Comment by rkrisztian** [comment]
Sorry, I was mistaken. I did not realize the rule is fine but I had a file name mismatch.
